www/js/controller.js

```diff
@@ -807,20 +782,36 @@ define([
 		/* Stub to update connections in graph */
 		addQuantity: function(source, destinations){
 		},
+		handleVariableType: function(e){
+			// Summary : Sets variableType to Unknown/Parameter/Dynamic
+			// Value is not allowed when variableType is Unknown
+			// Value is handled when variableType is parameter or dynamic.
+			console.log("********************* in handleVariableType");
+			var _variableType = e.target.value;
+			if(_variableType === this._model.active.getVariableType(this.currentID)){
+				return;
+			}
+			this.variableTypeControls(this.currentID, _variableType);
+			if(this._mode !== "AUTHOR"){
+				this.applyDirectives(this._PM.processAnswer(this.currentID, 'variableType', _variableType));
```

Comment:
there should have been a PM for author as well. Which would decide what to say. Even if it was as simple as showing done or not done. But I guess we are too late for that. Anyways I will try to refactor this at some point of time.